src/components/Network/NetworkTextual/NetworkTextual.tsx

```diff
@@ -0,0 +1,210 @@
+        (acc, val) => {
+            if (!acc.includes(val.source as string)) {
+                acc.push(val.source as string)
+            }
+            if (!acc.includes(val.target as string)) {
+                acc.push(val.target as string)
+            }
+            return acc
+        },
+        []
+    )
+
+    const objectsWithARelation = graphData?.nodes
+        ?.filter(node =>
+            objectUUIDSWithARelation?.includes(node.UUID as string)
+        )
+        .sort((a, b) => (a.Titel as string).localeCompare(b.Titel as string))
+
+    const filteredObjects = objectsWithARelation
+        ?.filter(obj => filters[obj.Type as string] === true)
+        .filter(obj =>
+            (obj.Titel?.toLowerCase() as string).includes(
+                filterQuery.toLowerCase()
+            )
+        )
+
+    const filteredObjectsLength = filteredObjects?.length
+
+    useEffect(() => {
+        // Get a list of the objects that have a relation with the selected object
+        const objectsWithRelationUUIDS = graphData?.links
+            ?.filter(
+                link =>
+                    link.source === selectedObject?.UUID ||
+                    link.target === selectedObject?.UUID
+            )
+            .map(link => {
+                if (link.source === selectedObject?.UUID) {
+                    return link.target
+                }
+                return link.source
+            })
+
+        const objectsWithRelation = graphData?.nodes?.filter(node =>
+            objectsWithRelationUUIDS?.includes(node.UUID as string)
+        )
+
+        setSelectedRelations(objectsWithRelation)
+    }, [selectedObject, graphData])
+
+    if (!graphData) return null
+
+    return (
+        <div className="pb-16">
+            <SelectedObjModal
+                selectedRelations={selectedRelations}
+                setSelectedObject={setSelectedObject}
+                selectedObject={selectedObject}
+            />
+
+            <div className="flex w-full mt-4 mb-2">
+                <input
+                    autoComplete="off"
+                    className={`block w-full pl-4 pr-24 md:pr-4 pt-2 pb-1 text-gray-700 border border-pzh-gray-400 rounded appearance-none focus:outline-none`}
+                    id="network-graph-search-query"
+                    placeholder={
+                        isMobile ? 'Filter' : 'Filter op titel van beleid'
+                    }
+                    type="text"
+                    value={filterQuery}
+                    onChange={event => setFilterQuery(event.target.value)}
+                />
+                {children}
+            </div>
+            <Text type="body-small">
+                {filteredObjectsLength === 1
+                    ? `Er is 1 resultaat gevonden`
+                    : `Er zijn ${filteredObjectsLength} resultaten gevonden`}
+            </Text>
+
+            <div className="grid grid-cols-6 px-4 mt-6">
```

Comment:
We have a Table component in @pzh-ui repo. Might be better to use that instead of buttons? You can check the MoLo as an example